ansible/roles/compute_init/tasks/main.yml

```diff
@@ -0,0 +1,150 @@
+    owner: root
+    group: root
+    mode: 0644
+  loop:
+    - ../../basic_users/library/terminate_user_sessions.py
+    - ../../stackhpc.os-manila-mount/library/os_manila_share.py
+    - ../../stackhpc.openhpc/library/sacct_cluster.py
+
+- name: Inject filter_plugins
+  copy:
+    src: '{{ item }}'
+    dest: '/etc/ansible-init/filter_plugins/{{ item | basename }}'
+    owner: root
+    group: root
+    mode: 0644
+  loop:
+    - ../../basic_users/filter_plugins/filter_keys.py
+    - ../../stackhpc.openhpc/filter_plugins/slurm_conf.py
+
+- name: Add filter_plugins ansible.cfg
+  lineinfile:
+    path: /etc/ansible-init/ansible.cfg
+    line: "filter_plugins = /etc/ansible-init/filter_plugins"
+    state: present
+    owner: root
+    group: root
+    mode: 0644
+
+- name: Ensure nfs /exports/cluster configured
+  block:
+    - name: Ensure the /exports/cluster directory exists
+      file:
+        path: /exports/cluster
+        state: directory
+        owner: root
+        group: root
+        mode: 0755
+
+    - name: Copy /etc/hosts to /exports/cluster
+      copy:
+        src: /etc/hosts
+        dest: /exports/cluster/hosts
+        owner: root
+        group: root
+        mode: 0644
+        remote_src: true
+
+    - name: Copy manila share info to /exports/cluster
+      copy:
+        content: "{{ os_manila_mount_share_info | to_nice_yaml }}"
+        dest: "/exports/cluster/manila_share_info.yml"
+      when: os_manila_mount_share_info is defined
+      
+    - name: Copy manila mount shares to /exports/cluster
+      copy:
+        content: "{{ os_manila_mount_shares | to_nice_yaml }}"
+        dest: "/exports/cluster/manila_shares.yml"
+      when: os_manila_mount_shares is defined
+
+    - name: Ensure /exports/cluster/cvmfs directory exists
+      file:
+        path: /exports/cluster/cvmfs
+        state: directory
+        owner: root
+        group: root
+        mode: 0755
+
+    - name: Copy EESSI CVMFS config to /exports/cluster
+      copy:
+        src: /etc/cvmfs/default.local
+        dest: /exports/cluster/cvmfs/default.local
+        owner: root
+        group: root
+        mode: 0644
+        remote_src: true
+
+    - name: Write openhpc munge key
+      copy:
+        content: "{{ vault_openhpc_mungekey | b64decode }}"
+        dest: "/exports/cluster/openhpc_munge.key"
+        owner: munge
+        group: munge
+        mode: 0400
+
+    # - name: Ensure /exports/cluster/inventory_hostname directory exists
+    #   file:
+    #     path: /exports/cluster/{{ inventory_hostname }}
+    #     state: directory
+    #     owner: root
+    #     group: root
+    #     mode: 0755
+
+    # - name: Template hostvars
+    #   template:
+    #     src: ../templates/hostvars.j2
+    #     dest: "/exports/cluster/{{ inventory_hostname }}/hostvars.yml"
+    #     owner: root
+    #     group: root
+    #     mode: 0644
+
+  delegate_to: "{{ groups['control'] | first }}"
```

Comment:
I think the logic here needs breaking up:

1. Stuff which only actually needs control host (e.g. coping /etc/hosts into nfs exports dir): Yes, having the play hosts as `compute_init` and delegating to first control make sense so the `compute_init` group becomes a functionality toggle. But it needs a `run_once: true` else e.g. with 100 hosts you're coping out those files 100 times over each other, and probably at some point that goes wrong.

2. Stuff which is actually host-specific, e.g. templating hostvars into the nfs export, you also want to run on the `compute_init` group but delegated to first control is right, but it needs to run on every host.